keel-sql/src/test/kotlin/com/netflix/spinnaker/keel/sql/SqlResourceRepositoryTests.kt

```diff
@@ -28,4 +42,58 @@ internal object SqlResourceRepositoryTests : ResourceRepositoryTests<SqlResource
   fun shutdown() {
     jooq.close()
   }
+
+  fun parallelCheckingTests() = rootContext<CheckLockFixture<SqlResourceRepository>> {
+    fixture {
+      CheckLockFixture(
+        subject = factory(Clock.systemDefaultZone())
+      )
+    }
+
+    after { flush() }
+
+    /**
+     * I'd like to have this test in the superclass but the in-memory implementation is not designed
+     * to be thread safe and the Redis implementation needs too much work to make it thread safe
+     * given it's not our preferred target.
+     */
+    context("many threads are checking simultaneously") {
```

Comment:
I think this satisfies me that we won't see the same resources getting selected for checking by multiple instances in the Keel cluster.